cmd/solvercli/cmd/balances.go

```diff
@@ -0,0 +1,331 @@
+}
+
+var balancesCmd = &cobra.Command{
+	Use:     "balances",
+	Short:   "Show current on-chain balances for USDC, gas tokens, and custom assets across all configured chains.",
+	Long:    `Show current on-chain balances for USDC, gas tokens, and custom assets across all configured chains.`,
+	Example: `solver balances --custom-assets '{"osmosis-1":["uosmo","uion"],"celestia-1":["utia"]}'`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := setupContext(cmd)
+
+		usdcBalances := make(map[string]*ChainBalance)
+		gasBalances := make(map[string]*ChainGasBalance)
+		customBalances := make(map[string][]*ChainBalance)
+		totalUSDCBalance := new(big.Int)
+		totalCustomAssetsUSDValue := new(big.Float)
+		err := getBalances(ctx, usdcBalances, gasBalances, customBalances, totalUSDCBalance, totalCustomAssetsUSDValue, cmd)
+		if err != nil {
+			lmt.Logger(ctx).Fatal("Failed to get balances", zap.Error(err))
+		}
+
+		fmt.Println("\nOn-Chain Balances:")
+		fmt.Println("------------------")
+
+		for chainID, usdc := range usdcBalances {
+			gas := gasBalances[chainID]
+			fmt.Printf("\nChain: %s\n", chainID)
+			fmt.Printf("  USDC Balance: %s USDC\n", normalizeBalance(usdc.Balance, CCTP_TOKEN_DECIMALS))
+			fmt.Printf("  Gas Balance: %s %s\n", normalizeBalance(gas.Balance, gas.Decimals), gas.Symbol)
+
+			if gas.Balance.Cmp(gas.CriticalThreshold) < 0 {
+				fmt.Printf("  ⚠️  Gas balance below critical threshold!\n")
+			} else if gas.Balance.Cmp(gas.WarningThreshold) < 0 {
+				fmt.Printf("  ⚠️  Gas balance below warning threshold\n")
+			}
+
+			// Print custom assets if available
+			if assets, ok := customBalances[chainID]; ok {
+				for _, asset := range assets {
+					fmt.Printf("  %s Balance: %s %s (%.2f USD)\n",
+						asset.AssetDenom,
+						normalizeBalance(asset.Balance, asset.Decimals),
+						asset.AssetDenom,
+						asset.ValueUSD)
+				}
+			}
+		}
+
+		fmt.Printf("\nTotals Across All Chains:")
+		fmt.Printf("\n------------------------\n")
+		fmt.Printf("Total USDC Balance: %s USDC\n", normalizeBalance(totalUSDCBalance, CCTP_TOKEN_DECIMALS))
+		if totalCustomAssetsUSDValue.Cmp(big.NewFloat(0)) > 0 {
+			fmt.Printf("Total Custom Assets USD Value: %.2f USD\n", totalCustomAssetsUSDValue)
+		}
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(balancesCmd)
+	balancesCmd.Flags().String("custom-assets", "", "JSON map of chain IDs to denom arrays, e.g. '{\"osmosis\":[\"uosmo\",\"uion\"],\"celestia\":[\"utia\"]}'")
+	balancesCmd.Flags().String("evm-address", "", "Optional EVM address to check balances for instead of config address")
+	balancesCmd.Flags().String("osmosis-address", "", "Optional Osmosis address to check balances for instead of config address")
+}
+
+func getBalances(ctx context.Context,
+	usdcBalances map[string]*ChainBalance,
+	gasBalances map[string]*ChainGasBalance,
+	customBalances map[string][]*ChainBalance,
+	totalUSDCBalance *big.Int,
+	totalCustomAssetsUSDValue *big.Float,
+	cmd *cobra.Command) error {
```

Comment:
nit, should be this for consistency 
```suggestion
func getBalances(
        ctx context.Context,
	usdcBalances map[string]*ChainBalance,
	gasBalances map[string]*ChainGasBalance,
	customBalances map[string][]*ChainBalance,
	totalUSDCBalance *big.Int,
	totalCustomAssetsUSDValue *big.Float,
	cmd *cobra.Command
) error {
```